python/lsst/ts/ofc/scripts/generate_sensitivity_matrix.py

```diff
@@ -0,0 +1,393 @@
+
+    # Convert to angles
+    thx = radii * np.cos(azs)
+    thy = radii * np.sin(azs)
+
+    # Compute Zernike coefficients
+    coefs = []
+    for thx_, thy_ in zip(thx, thy):
+        coefs.append(
+            batoid.zernike(optic, thx_, thy_, wavelength, jmax=jmax, eps=0.61, nx=255)
+        )
+    coefs = np.array(coefs)
+
+    basis = galsim.zernike.zernikeBasis(kmax, thx, thy, R_outer=field)
+    dzs = np.dot(basis, coefs * w[:, None]) / np.pi
+
+    return dzs
+
+
+def get_intrinsic_zk(
+    config: dict,
+    instrument: str,
+    band: str,
+    jmax: int = 22,
+    kmax: int = 15,
+    rings: int = 11,
+    spokes: int = 35,
+) -> np.ndarray:
+    """Get the intrinsic double zernike coefficients
+    for a given instrument and band.
+
+    Parameters
+    ----------
+    config : dict
+        Configuration dictionary.
+    instrument : str
+        Name of the instrument. Either 'LSST' or 'ComCam'.
+    band : str
+        Name of the filter. Either 'u', 'g', 'r', 'i', 'z', or 'y'.
+    jmax : int, optional
+        Maximum j index, by default 22
+    kmax : int, optional
+        Maximum k index, by default 15
+    rings : int, optional
+        Number of rings in the field, by default 11
+    spokes : int, optional
+        Number of spokes in each ring, by default 35
+
+    Returns
+    -------
+    dz0 : np.ndarray
+        Intrinsic double zernike coefficients.
+    """
+
+    fiducial, wavelength = get_fiducial(instrument, band)
+
+    return double_zernike(
+        fiducial,
+        field=np.deg2rad(config["pupil"]["R_outer"]),
+        wavelength=wavelength * 1e-6,  # nm -> m
+        eps=0.61,
+        rings=rings,
+        spokes=spokes,
+        jmax=jmax,
+        kmax=kmax,
+    )
+
+
+def get_sensitivity_dz(
+    config: dict,
+    instrument: str,
+    band: str,
+    ndof: int = 50,
+    jmax: int = 22,
+    kmax: int = 30,
+) -> np.ndarray:
+    """Get the double zernike sensitivity matrix
+    for a given instrument and band.
+
+    Parameters
+    ----------
+    config : dict
+        Configuration dictionary.
+    instrument : str
+        Name of the instrument. Either 'LSST' or 'ComCam'.
+    band : str
+        Name of the filter. Either 'u', 'g', 'r', 'i', 'z', or 'y'.
+    ndof : int, optional
+        Number of degrees of freedom, by default 50
+    jmax : int, optional
+        Maximum j index, by default 22
+    kmax: int, optional
+        Maximum k index, by default 30
+
+    Returns
+    ----------
+    sensitivity_matrix : np.ndarray
+        Double zernike sensitivity matrix.
+    """
+
+    sensitivity_matrix = np.empty((ndof, kmax + 1, jmax + 1))
+    # Get fiducial telescope
+    fiducial, wavelength = get_fiducial(instrument, band)
+
+    # Compute the intrinsic double zernike coefficients
+    dz0 = get_intrinsic_zk(config, instrument, band, jmax, kmax)
+
+    # Swap the zernikes that are flipped in batoid
+    swap = [2, 5, 8, 10, 13, 15, 16, 18, 20]
+    dz0[:, swap] *= -1
```

Comment:
Is there any ticket to resolve this mapping?

Reply:
Yes, see above